core/src/test/java/com/linecorp/armeria/server/ServerTlsTest.java

```diff
@@ -0,0 +1,90 @@
+import javax.net.ssl.TrustManagerFactory;
+
+import org.junit.jupiter.api.Test;
+
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+
+class ServerTlsTest {
+
+    @Test
+    void testJksKeyStoreWithNullPassword() throws Exception {
+        /*
+         * Dummy keystore generation
+         * keytool -genkeypair -keyalg RSA -keysize 2048 -storetype JKS -keystore keystore.jks -validity 3650
+         * key store password = password
+         * key password = keypassword
+         */
+        final String keyPass = null;
+        final KeyStore keyStore = KeyStore.getInstance("JKS");
+        keyStore.load(getClass().getResource("keystore.jks").openStream(),
+                      keyPass != null ? keyPass.toCharArray() : null);
+
+        final KeyManagerFactory kmf = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+        kmf.init(keyStore, "keypassword".toCharArray());
+
+        final TrustManagerFactory tmf
+                = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+        tmf.init(keyStore);
+
+        final Server server
+                = new ServerBuilder().service("/", (ctx, res) -> HttpResponse.of(HttpStatus.OK))
+                                     .tls(kmf, sslContextBuilder -> {
+                                         sslContextBuilder.keyManager(kmf);
+                                         sslContextBuilder.trustManager(tmf);
+                                     })
+                                     .build();
+    }
+
+    @Test
+    void testPkcs12KeyStoreWithNullPassword() throws Exception {
+        // Dummy keystore generation
+        // keytool -genkeypair -keyalg RSA -keysize 2048 -storetype PKCS12 -keystore keystore.p12 -validity 3650
+        // keypassword = keystorepassword = password
+        assertThrows(RuntimeException.class, () -> {
+            final String keyPass = null;
+            final KeyStore keyStore = KeyStore.getInstance("PKCS12");
+            keyStore.load(getClass().getResource("keystore.p12").openStream(),
+                          keyPass != null ? keyPass.toCharArray() : null);
+
+            final KeyManagerFactory kmf
+                    = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+            kmf.init(keyStore, "password".toCharArray());
+
+            final TrustManagerFactory tmf
+                    = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+            tmf.init(keyStore);
+
+            final Server server
+                    = new ServerBuilder().service("/", (ctx, res) -> HttpResponse.of(HttpStatus.OK))
+                                         .tls(kmf, sslContextBuilder -> {
+                                             sslContextBuilder.keyManager(kmf);
+                                             sslContextBuilder.trustManager(tmf);
+                                         })
```

Comment:
I don't think you need to set the `TrustManagerFactory` unless you want to authenticate clients. You can just pass a no-op customizer, e.g. `sslContextBuilder -> {}`.

Reply:
Fixed it. Thank you.